src/NServiceBus.Core/Hosting/HostCreator.cs

```diff
@@ -90,5 +95,15 @@ public static async Task<IStartableEndpoint> CreateWithInternallyManagedContaine
 
             return new InternallyManagedContainerHost(startableEndpoint, hostingComponent);
         }
+
+        static void CheckIfSettingsWhereUsedToCreateAnotherEndpoint(SettingsHolder settings)
+        {
+            if (settings.GetOrDefault<bool>("UsedToCreateEndpoint"))
+            {
+                throw new ArgumentException("This EndpointConfiguration was already used for starting an endpoint, create a new one before starting the other.");
```

Comment:
> create a new one before starting the other

is vague and easy to misinterpret. Maybe something like this: 

```suggestion
                throw new ArgumentException("This EndpointConfiguration was already used for starting an endpoint. Each endpoint requires a new EndpointConfiguration.");
```